### PR TITLE
Harden choice modality discriminator and keyboard commit paths

### PR DESCRIPTION
**Pull request by The-Obstacle-Is-The-Way**
## Problem

A pre-promotion adversarial review of the merged BUG-274 wave (5 lenses, per-finding verification, all findings CONFIRMED against dev head `cb58133a` — one reproduced end-to-end in Chromium) surfaced three defects in the new interaction machinery before it reaches production:

1. **Stale pointer-arm resurrects BUG-274.** A right-click press fired `pointerdown` and armed the activation flag, but a secondary press never produces a `click`, `pointerleave` is guarded on held buttons, and no expiry existed — so the arm survived until a later arrow-key selection landed on that choice and auto-committed it. An Alt-Tab-abandoned press leaked the same way.
2. **Enter inside choice markdown hijacked links.** The fieldset-level Enter handler committed for Enter on ANY focusable descendant — following a reference link inside a choice graded the answer.
3. **Both keyboard commit paths dropped focus to `<body>`.** The Submit button unmounts the instant `isPending` flips and the radios become disabled when feedback arrives, stranding keyboard/screen-reader users exactly as the verdict renders — undermining the wave's own accessibility goal.

## Solution (red-first per concern)

- `d31434b0`: primary-button-only arming + contextmenu clear + window-blur clear. Red specs: right-click-then-ArrowUp misclassification and blur-then-AT-click both failed before the fix.
- `3c00f8c0`: Enter commits only from the radio inputs; links keep native behavior. Red spec: link-focus Enter failed before the fix; radio-Enter pin green throughout.
- `a9c3a1f6`: the feedback region is focusable (tabIndex −1, canonical outline suppression) and receives focus in the same effect that scrolls it into view — extending the session-summary focus precedent; `focusElementWithoutScroll` promoted to the shared practice components directory (3 imports updated). Red spec pins `document.activeElement` on the feedback region after commit. Also adds the missing orchestrator seam test (PracticeSessionPageView `canSubmit`/`onSubmit` pass-through — previously deletable without any test failing) and records accepted known issue AF-8 (selected-uncommitted choice discarded by pre-feedback navigation) in interaction-contracts.md.

Considered and deliberately not changed: SessionSummaryView focusing its heading on passive direct-URL loads is standard SPA view-load focus practice, not a defect.

## Verification

Full gate on final head: typecheck, lint, unit 3136/3136, browser 357/357 (5 new specs), integration 154 passed / 2 skipped, build, E2E 36/36.

https://claude.ai/code/session_01AND8DhkyKL44snLEXW2wq1

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved focus handling after submitting practice answers so keyboard focus stays on the feedback area instead of jumping away.
  * Made Enter key behavior safer: it now submits only from the selected radio choice, not from links or other interactive content inside answers.
  * Fixed choice selection so mouse, keyboard, and assistive tech interactions are classified more reliably.

* **Tests**
  * Added browser coverage for submit behavior, focus movement, and keyboard interaction edge cases.

* **Documentation**
  * Clarified tutor-mode interaction behavior and noted a known navigation limitation.

### CODE REVIEW

File: app/(app)/app/practice/components/practice-view.browser.spec.tsx [hunk 1]
```diff
@@ -797,3 +797,88 @@ test('calls onEndSession from the bottom-bar End session button on the last tuto
...
+  const feedbackRegion = screen.getByTestId('answer-feedback-region');
+  await expect.element(feedbackRegion).toBeInTheDocument();
+  await vi.waitFor(() => {
+    if (document.activeElement !== feedbackRegion.element()) {
+      throw new Error(
+        `Expected focus on the feedback region, got ${document.activeElement?.tagName}`,
+      );
+    }
+  });
+});
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply app/(app)/app/practice/components/practice-view.browser.spec.tsx]: Fixed in f59b627f: harness + test extracted to practice-view-commit-focus.browser.spec.tsx; the main spec is back to 799 lines and the Biome warning is gone.

[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply app/(app)/app/practice/components/practice-view.browser.spec.tsx]: Fixed in f59b627f: replaced the vi.waitFor polling with await expect.element(feedbackRegion).toHaveFocus().
